cmd/cp-main.go

```diff
@@ -57,9 +58,16 @@ var (
 			Name:  "encrypt",
 			Usage: "encrypt/decrypt objects (using server-side encryption with server managed keys)",
 		},
+		cli.StringFlag{
+			Name:  "attr",
+			Usage: "add custom metadata for the object",
+		},
 	}
 )
 
+// ErrInvalidMetadata reflects invalid metadata format
+var ErrInvalidMetadata = probe.NewError(errors.New("specified metadata should be of form key1=value1,key2=value2,... and so on"))
```

Comment:
This is not a correct style of error when using probe package for example look at errDummy() implementation, you should use that style. 

Using probe in this manner adds the line number of this definition meaning mc --debug will show error is originating here which is not we need to stop at the function return.

Reply:
Also the error to point to the exact position to indicate the error, basically use the input Arg and show in which position you see this problem. If that's not possible we can simply show the entire arg and indicate it doesn't follow expected style.

Reply:
OK, so can we have, `var ErrInvalidMetadata = errors.New("the appropriate error message")` and call `probe.NewError(ErrInvalidMetadata)` at the respective call sites?

Reply:
yes that is possible @krisis